development/image-detector/internal/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,68 @@
+package kubernetes
+
+import (
+	"errors"
+	"io"
+	"os"
+
+	"gopkg.in/yaml.v3"
+)
+
+type DeploymentFile struct {
+	Spec SpecField `yaml:"spec"`
+}
+
+type SpecField struct {
+	Template Template `yaml:"template"`
+}
+
+type Template struct {
+	Spec PodSpec `yaml:"spec"`
+}
+
+type PodSpec struct {
+	Containers []Container `yaml:"containers"`
+}
+
+type Container struct {
+	Image string `yaml:"image"`
+}
+
+func Extract(path string) ([]string, error) {
```

Comment:
Please change function name to more descriptive. From package and function name it's not clear what it extracts. There is no comments for it. This is the same for other files and functions where you name function like estractPostsubmit, but you extract images only not the whole postsubmits.

Reply:
Renamed and moved to separate package make it something like `extractimage.Fromxxx()` e. g. `extratimages.FromProwJobs(path)`